### PR TITLE
DUPLO-18748 feat: add support for healthcheck configuration for Duplo service lbconfigs

### PR DESCRIPTION
**Pull request by Matheusbafutto**
## Overview

This PR introduces healthcheck configurations for load balancer configurations from a duplo service

## Summary of changes

This PR does the following:

- Adds support for setting advanced healthcheck configurations on a service's load balancer configuration

## Testing performed

- [ ] Using unit tests
- [x] Manually, on my local system
- [ ] Manually, on a remote test system

## Describe any breaking changes

Load balancer configurations have a particularly wide range of use cases which makes effective sanity checking difficult. QA  verification recommended

___

### **PR Type**
Enhancement, Documentation

___

### **Description**
- Introduced a new `health_check` configuration block to the load balancer schema in `resource_duplo_lbconfigs.go`.
- Implemented logic to handle the new health check parameters in the resource creation and update functions.
- Flattened the health check configuration for easier state management.
- Updated the `CHANGELOG.md` to document the addition of advanced health check configurations.
- Added detailed documentation for the new `health_check` configuration block in `docs/resources/duplo_service_lbconfigs.md`.
- Provided an example usage of the new `health_check` configuration block in `examples/resources/duplocloud_duplo_service_lbconfigs/resource.tf`.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>resource_duplo_lbconfigs.go</strong><dd><code>Add health check configuration to load balancer schema</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

duplocloud/resource_duplo_lbconfigs.go

<li>Added a new <code>health_check</code> configuration block to the load balancer <br>schema.<br> <li> Implemented logic to handle the new health check parameters in the <br>resource creation and update functions.<br> <li> Flattened the health check configuration for easier state management.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/590/files#diff-41a83bf1ecdb965c35671829b7fa60279941ef1a968e3410c2b9e2f285162e10">+76/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>CHANGELOG.md</strong><dd><code>Update changelog with health check configuration enhancement</code></dd></summary>
<hr>

CHANGELOG.md

<li>Documented the addition of advanced health check configurations for <br>load balancer configurations.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/590/files#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4ed">+5/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>duplo_service_lbconfigs.md</strong><dd><code>Document health check configuration for load balancers</code>&nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

docs/resources/duplo_service_lbconfigs.md

<li>Added documentation for the new <code>health_check</code> configuration block.<br> <li> Detailed the parameters available within the <code>health_check</code> block.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/590/files#diff-1948136dba5174d211a7b5c40cd528de0c742563deae48e7df4bd31bd14ef307">+22/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>resource.tf</strong><dd><code>Add example for health check configuration in Terraform</code>&nbsp; &nbsp; </dd></summary>
<hr>

examples/resources/duplocloud_duplo_service_lbconfigs/resource.tf

<li>Added example usage of the new <code>health_check</code> configuration block in a <br>Terraform resource.<br>

</details>

  </td>
  <td><a href="https://github.com/duplocloud/terraform-provider-duplocloud/pull/590/files#diff-2d05234bc48520ea10772efb1631da6acf92812966e68857517c67a383cde6b6">+8/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-18748 [TF] Support editing TG properties on duplocloud_duplo_service_lbconfigs](https://app.clickup.com/t/8655600/DUPLO-18748)

File: duplocloud/resource_duplo_lbconfigs.go [hunk 1]
```diff
@@ -493,6 +556,14 @@ func duploServiceLbConfigsWaitUntilReady(ctx context.Context, c *duplosdk.Client
 
 func flattenDuploServiceLbConfiguration(lb *duplosdk.DuploLbConfiguration) map[string]interface{} {
 	log.Printf("[DEBUG] flattenDuploServiceLbConfiguration... Start")
+	healthcheckConfig := map[string]interface{}{
```
Comment (tahir-duplo) [inline duplocloud/resource_duplo_lbconfigs.go:559]: Can we please add nil check `lb.HealthCheckConfig` to avoid nil pointer exception?

**Review by Duplo-nikv — DISMISSED** [review]
LGTM

**Review by tahir-duplo — DISMISSED** [review]
LGTM

**Review by tahir-duplo — APPROVED** [review]
LGTM

**Pull request merged**